### ISSUE TITLE
jobs table entries for backfills should include shard info

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#5020]
The current `rethinkdb.jobs` entry for backfills should have an indication of what shard(s) it is working on. The current output format is:
```
{
  'duration_sec': 0.059261,
  'id': ['backfill', '195e49e2-cf0d-53b8-81e3-1a8f3a7bfe55'],
  'info': {
    'db': 'test',
    'destination_server': 'server_1',
    'progress': 0,
    'source_server': 'server_0',
    'table': 'backfills_py2_7_10'
  },
  'servers': ['server_0', 'server_1'],
  'type': 'backfill'
}
```

**Comment by danielmewes** [comment]
It's not obvious how shards should be represented in the entries. Though we could use their index into the `table_config` shard array.

There's also the question of how useful this will be for users. For most use cases of the `jobs` table the shards probably don't matter.
I think you need this data for the backfill test though, right?
An alternative source for information about ongoing backfills on a per-shard basis is the `table_status` table, where shards that are backfilling will be in the `"backfilling"` state.

**Comment by VeXocide** [comment]
@larkost did initially ask for this due to the backfill test, though, after having given it some thought, I'm not convinced of its usefulness to users either. Using the index into the `table_config` shard array sounds like a bad idea due to the config potentially in between, not sure how else we could represent it though.

**Comment by larkost** [comment]
My proximate use case is of course the tests that I am writing. I realized that I could not verify that a given backfill entry in the `rethinkdb.jobs` table was the one I just caused. The `table_status` information would be a second circumstantial bit of evidence, but still does not directly connect one to the other. This is definitely not a blocker, but does seem a missing piece.

As to the usefulness to users, the one that comes to mind would be someone building a reconfiguration tool. If it was doing something complicated to live-migrate among multiple servers it would be a lot easier to write if it had better insight into what was done and what was not.

On the subject of how to expose this, I don't have any ideas better than adding a uuid, and that feels a big heavyweight.
